4.gene-expression-signatures/scripts/2.run_gsea.py

```diff
@@ -161,44 +167,56 @@ def perform_gsea(weights_df: pd.DataFrame, model_name: str, num_components: int,
     combined_results_df = pd.DataFrame()
     print(f"No existing file found. Initialized empty DataFrame.")
 
-for num_components in latent_dims:
-    for model_name in model_names:
-        # Load the saved model
-        # Check if this model and latent dimension have already been processed
-        if not combined_results_df.empty:
-            if ((combined_results_df['model'] == model_name) & 
-                (combined_results_df['full_model_z'] == num_components)).any():
-                print(f"Skipping {model_name} with {num_components} dimensions as it is already processed.")
-                continue  # Skip to the next iteration if this combination is already present
-        model_filename = model_save_dir / f"{model_name}_{num_components}_components_model.joblib"
-        if model_filename.exists():
-            print(f"Loading model from {model_filename}")
-            model = joblib.load(model_filename)
-
-            # Extract the weight matrix
-            try:
-                weight_matrix_df = extract_weights(model, model_name, weight_data)
-            except ValueError as e:
-                print(e)
-                continue
-
-            # Perform GSEA
-            gsea_results_df = perform_gsea(weight_matrix_df, model_name, num_components)
-            combined_results_df = pd.concat([combined_results_df, gsea_results_df], ignore_index=True)
-        else:
-            print(f"Model file {model_filename} not found. Skipping.")
 
+# Iterate over all files in the saved_models directory
+for model_file in model_save_dir.glob("*.joblib"):
+    # Extract model name and number of components from the filename
+    model_file_name = model_file.stem
+    try:
+        # Assuming the filename format includes model_name, num_components, and potentially a seed
+        # Example: "BetaVAE_100_components_seed42_model.joblib"
+        parts = model_file_name.split("_")
+        model_name = parts[0]  # First part is the model name
+        num_components = int(parts[3])  # Second part should indicate the number of components
+        init = int(parts[7])
+        seed = int(parts[9])
+    except (IndexError, ValueError) as e:
+        print(f"Skipping file {model_file} due to unexpected filename format.")
+        continue
+    # Check if this model, latent dimension, and initialization have already been processed
+    if not combined_results_df.empty:
+        if ((combined_results_df['model'] == model_name) & 
+            (combined_results_df['init'] == init) &
+            (combined_results_df['full_model_z'] == num_components)).any():
+            print(f"Skipping {model_name} init {init} with {num_components} dimensions as it is already processed.")
+            continue
```

Comment:
I like these continue statements in principle, but it could be dangerous, especially if you're underlying data changes. You do have a print statement above to catch these sorts of potential errors, so they won't be completely silent. You could, alternatively, consider adding a "force" argument that will rewrite if you explicitly want it too